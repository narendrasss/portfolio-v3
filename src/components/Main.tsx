import React from 'react';
import { Flex, FlexProps } from 'rebass';
import styled from 'styled-components';

const Container = styled(Flex)`
  width: 100vw;
  height: 100vh;
` as React.FC<FlexProps>;

const Main: React.FC<FlexProps> = props => (
  <Container as="main" flexDirection="column" {...props} />
);

export default Main;
