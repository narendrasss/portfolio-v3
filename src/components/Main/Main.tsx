import React from 'react';
import { Flex, FlexProps } from 'rebass';
import styled from 'styled-components';

export { FlexProps as MainProps } from 'rebass';

const Container = styled(Flex)`
  width: 100vw;
  min-height: 100vh;
` as React.FC<FlexProps>;

const Main: React.FC<FlexProps> = props => (
  <Container as="main" flexDirection="column" {...props} />
);

export default Main;
