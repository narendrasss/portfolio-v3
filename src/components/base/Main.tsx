import styled from 'styled-components';
import {
  justifyContent,
  alignItems,
  JustifyContentProps,
  AlignItemsProps,
} from 'styled-system';

type MainProps = JustifyContentProps & AlignItemsProps;

const Main = styled.main<MainProps>`
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  padding: 3rem;
  ${justifyContent}
  ${alignItems}
`;

export default Main;
