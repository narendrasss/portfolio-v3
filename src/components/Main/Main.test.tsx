import React from 'react';
import { render } from 'react-testing-library';
import Main from './Main';

describe('Main', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Main />);
    expect(asFragment()).toMatchSnapshot();
  });
});
