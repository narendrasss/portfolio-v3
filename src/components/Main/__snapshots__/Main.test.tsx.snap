// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders without crashing 1`] = `
<DocumentFragment>
  <main
    class="sc-bdVaJa sc-bwzfXH sc-htoDjs jiVnaf"
  />
</DocumentFragment>
`;
